File: algorithms/B/Beautiful Arrangement II/Beautiful Arrangement II.cpp
class Solution {
public:
    vector<int> constructArray(int n, int k) {
        vector<int> nums = {1}; 
        int st = 1, last = 1, t = k;
        while(k){
            last = last+k*st; 
            nums.push_back(last); st = st*-1; k--;
        }
        for(int i = t+2; i<=n;++i) nums.push_back(i);
        return nums;
    }
};

File: algorithms/B/Beautiful Arrangement II/Beautiful Arrangement II.java
class Solution {
    public int[] constructArray(int n, int k) {
        int [] result = new int[n];
        result[0] = 1;
        int sign = 1;
        for(int i = 1 ; i < n; i++, k--){
            if(k > 0){
                result[i] = result[i-1] + k * sign;
                sign *= -1;
            }
            else{
                result[i] = i+1;
            }
        }
        return result;
    }
}
